backend.go

```diff
@@ -54,16 +59,100 @@ func backend() *azureAuthBackend {
 		},
 		Paths: framework.PathAppend(
 			[]*framework.Path{
-				pathLogin(b),
-				pathConfig(b),
+				pathLogin(&b),
+				pathConfig(&b),
+				pathRotateRoot(&b),
 			},
-			pathsRole(b),
+			pathsRole(&b),
 		),
+		WALRollback:  b.walRollback,
+		PeriodicFunc: b.periodicFunc,
 	}
 
 	b.resourceAPIVersionCache = make(map[string]string)
 
-	return b
+	return &b
+}
+
+func (b *azureAuthBackend) periodicFunc(ctx context.Context, sys *logical.Request) error {
```

Comment:
I think it's worth adding a comment here (maybe some other locations too?) to note that the periodic function is responsible for eventually swapping out the root credential for rotation operations.

Reply:
Added some documentation for the function and added comments in places that it is being used